src/git/errors.ts

```diff
@@ -488,3 +488,68 @@ export class WorktreeDeleteError extends Error {
 		Error.captureStackTrace?.(this, WorktreeDeleteError);
 	}
 }
+
+export const enum TagErrorReason {
+	TagAlreadyExists,
+	TagNotFound,
+	InvalidTagName,
+	PermissionDenied,
+	RemoteRejected,
+	Other,
+}
+
+export class TagError extends Error {
```

Comment:
What do you think about either splitting this up into one class per tag action (create/delete/get) like we do above this with worktrees/stashes/etc. or passing in the action so that the message is more descriptive to the action being done to the tag?

Reply:
I'm not entirely sure if it's worth splitting into one class per action, as it may cause a lot of duplicated code. In fact, I wanted to revisit the errors once all git actions are done, because it may be interesting to have a more generic `GitError` class that receives information about the git command and action.
I agree on the passing the action as an argument to the constructor (and maybe even with a builder method as well). I'll make the changes, thanks for the feedback!